azure/src/wizard/StorageAccountCreateStep.ts

```diff
@@ -47,6 +51,57 @@ export class StorageAccountCreateStep<T extends types.IStorageAccountWizardConte
         ext.outputChannel.appendLog(createdStorageAccount);
     }
 
+    protected async generateRelatedName(wizardContext: T, name: string, namingRules: IAzureNamingRules | IAzureNamingRules[]): Promise<string | undefined> {
+        if (!isArray(namingRules)) {
+            namingRules = [namingRules];
+        }
+
+        let preferredName: string = namingRules.some((n: IAzureNamingRules) => !!n.lowercaseOnly) ? name.toLowerCase() : name;
+
+        for (let invalidCharsRegExp of namingRules.map((n: IAzureNamingRules) => n.invalidCharsRegExp)) {
+            // Ensure the regExp uses the 'g' flag to replace _all_ invalid characters
+            invalidCharsRegExp = new RegExp(invalidCharsRegExp, 'g');
+            preferredName = preferredName.replace(invalidCharsRegExp, '');
+        }
+
+        const minLength: number = Math.max(...namingRules.map((n: IAzureNamingRules) => n.minLength));
+        const maxLength: number = Math.min(...namingRules.map((n: IAzureNamingRules) => n.maxLength));
+
+        const maxTries: number = 100;
```

Comment:
100 attempts (which will include a web request to check the name availability) is an awful lot of tries. I think 3-5 is a more reasonable maximum.

You could use a sort of exponential backoff on the `i` value, or choose a random number between 1-X, and try that, rather than sequentially checking each `i`.

Reply:
I reduced the amount of trying times to 5. But I couldn't understand quite well your comment on checking each `i`. Could you please elaborate on it?

Reply:
What I mean is that as you have it implemented, you are checking 0, 1, 2, 3..., i.e. mystorageaccount2, mystorageaccount3, mystorageaccount4, etc. I think there are three reasonable approaches to this:

1. Sequential checking, like you are doing. As an advantage, it is "predictable", and behaves just like duplicate files are handled in OSs.
2. Exponential checking, e.g. you check 2^i (1, 2, 4, 8, 16, ...) or something similar
3. Random checking, e.g. pick a random number 1 to 100, check if it exists, if so just try again

I think it's sort of a PM question as to whether approach 1 is important enough to be desirable.

Reply:
Ohhh I see your point of view now. However I feel that sequential, int this case is not that bad as we are just doing it 5 times, random and exponential will be the same. Do you know who I can reach out to? or how can i solve this issue?

Reply:
I like the idea and I can't think of any immediately confusing or bad side-effects. I say go for it.

Reply:
Which one?

Reply:
I talked to @fiveisprime offline, the suggestion was to pick a random one 1-100, with a few retries (e.g. 5).